staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1368,22 +1369,32 @@ func (c *cacheWatcher) processEvents(ctx context.Context, initEvents []*watchCac
 	// consider increase size of result buffer in those cases.
 	const initProcessThreshold = 500 * time.Millisecond
 	startTime := time.Now()
-	for _, event := range initEvents {
+
+	initEventRcvd := 0
+	for {
+		event, err := cacheInterval.Next()
+		if err != nil {
+			klog.V(2).Info(err)
+			break
```

Comment:
This shouldn't be break right?

We probably should emit an Error event and return - please double check what would currently happen.

Reply:
BTW - this is this piece of logic that requires adding a new unit test.

Reply:
> We probably should emit an Error event and return - please double check what would currently happen.

@wojtek-t - quick question:
Just to make sure I understand why we want to emit an error event: we want to emit one because it signifies that the event which was supposed to have RV just greater than the one for which the watch started for, has been popped off and the watch should probably be re-established?

Reply:
Yeah - we can't proceed to serving some other events, because that could violate the watch invariant of eventual consistency (if some object changed there and doesn't change later). [Also the ordering.]

Alterntaively we can just stop the watch, but I think emiting an error is better.
We already have logic that should allow us to handle that:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/cache/reflector.go#L479
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/cache/reflector.go#L432
So that should be better.

That said - we need need a good test for it.

Reply:
Sounds good, thanks for the clarification. I'll get working on that :)